Update Snowflake Connector Example Doc (#454)
wso2/docs-mi: en/docs/reference/connectors/snowflake-connector/snowflake-connector-example.md

# Snowflake connector example

> The Snowflake Connector allows you to access the [Snowflake JDBC Driver API](https://docs.snowflake.com/developer-guide/jdbc/jdbc) from an integration sequence. This provides functionalities to execute a set of standard Snowflake DDL, DML, and query commands. You can use the connector to perform query, execute and batch execute operations on Snowflake databases. 

Reviewer: ```suggestion
The Snowflake Connector allows you to access the [Snowflake JDBC Driver API](https://docs.snowflake.com/developer-guide/jdbc/jdbc) from an integration sequence. This provides functionalities to execute a set of standard Snowflake DDL, DML, and query commands. You can use the connector to perform query, execute, and batch execute operations on Snowflake databases. 
```

## What you'll build

> This example demonstrates how to use the Snowflake Connector for querying, inserting, batch inserting, and deleting data in a Snowflake database. Let's consider a use case involving a hotel database for managing reservations. Within the hotel database, there exists a table named RESERVATIONS responsible for storing reservation details.

Reviewer: ```suggestion
This example demonstrates how to use the Snowflake Connector for querying, inserting, batch inserting, and deleting data in a Snowflake database. Let's consider a use case involving a hotel database for managing reservations. Within the hotel database, there is a table named RESERVATIONS that stores reservation details.
```

```sql
create table RESERVATIONS (
   NICNUMBER String,
   FIRSTNAME String,
   LASTNAME String,
   CHECKIN String,
   CHECKOUT String,
   ADULTS INT,
   CHILDREN INT,
   ROOMTYPE String,
   SPECIALREQUESTS String
)
```

If you do not want to configure this yourself, you can simply [get the project](#get-the-project) and run it.

> ### Setup the integration project

Reviewer: ```suggestion
### Set up the integration project
```

Create the following resources in the 'SnowflakeConnectorApi' REST API.

| uri-template               | method  |
|----------------------------|-------- |
| `/insertReservation`         | POST    |
| `/getReservationInfo`        | GET     |
| `/insertReservationBatch`    | POST    |
| `/deleteReservation/{NICNUMBER}`         | DELETE  |

Let's add the operations to the resources in the `SnowflakeConnectorApi` API.

#### - /insertReservation

> Users can utilize this resource to insert a single record into the Snowflake database. The user will be sending the reservation payload in the request body.

Reviewer: ```suggestion
Users can utilize this resource to insert a single record into the Snowflake database. The user will send the reservation payload in the request body.
```

      1. Select the Snowflake connection configuration you created.
      2. In the `Sql Query` text box, enter the following query.
      
         ```sql
         SELECT * FROM HOTEL_DB.PUBLIC.RESERVATIONS
         ```

         <img src="{{base_path}}/assets/img/integrate/connectors/snowflake_connector/snowflake-conn-query-config.png" title="Snowflake query configuration" width="400" alt="Snowflake query configuration"/>

         ```xml
         <snowflake.query configKey="SNOWFLAKE_CONNECTION">
            <query>SELECT * FROM HOTEL_DB.PUBLIC.RESERVATIONS</query>
         </snowflake.query>
         ```

2. Add the [Property Mediator]({{base_path}}/reference/mediators/property-mediator/) and set the Property name as `messageType` and the value as `application/json`. This is added so that the response will be in JSON.
   ```xml
    <property name="messageType" scope="axis2" type="STRING" value="application/json"/>
   ```
3. Add the [Respond Mediator]({{base_path}}/reference/mediators/respond-mediator/) to send back the response.

#### - /insertReservationBatch

> Using this resource users can insert multiple records into the table `Reservations` of `HOTEL_DB`. The user will be sending the bulk payload in the request body.

Reviewer: ```suggestion
Using this resource users can insert multiple records into the table `Reservations` of `HOTEL_DB`. The user will send the bulk payload in the request body.
```

         You should receive a response as below.
         ```json
         [
            {
               "NICNUMBER":"9876543210",
               "FIRSTNAME":"Alice",
               "LASTNAME":"Johnson",
               "CHECKIN":"2023-10-01",
               "CHECKOUT":"2023-10-05",
               "ADULTS":"1",
               "CHILDREN":"0",
               "ROOMTYPE":"Single",
               "SPECIALREQUESTS":"Quiet room"
            },
            {
               "NICNUMBER":"2345678901",
               "FIRSTNAME":"Emma",    
               "LASTNAME":"Williams",
               "CHECKIN":"2024-06-01",
               "CHECKOUT":"2024-06-05", 
               "ADULTS":"1",
               "CHILDREN":"0",
               "ROOMTYPE":"Single",
               "SPECIALREQUESTS":"Late check-in"
            },
            {
               "NICNUMBER":"1234567890",
               "FIRSTNAME":"Bob",
               "LASTNAME":"Smith",
               "CHECKIN":"2023-10-01",
               "CHECKOUT":"2023-10-05",
               "ADULTS":"2",
               "CHILDREN":"1",
               "ROOMTYPE":"Double",
               "SPECIALREQUESTS":"Extra bed"
            }
         ]

>          ```

Reviewer: Let's rename ## What's Next -> ## What's next
